omod/src/main/java/org/openmrs/module/coreapps/page/controller/MarkPatientDeadPageController.java

```diff
@@ -53,7 +63,12 @@ public String post(@SpringBean("patientService") PatientService patientService,
                 patient.setCauseOfDeath(null);
             }
 
+
             patientService.savePatient(patient);
+
+            if (patient.isDead() && patient.getDeathDate() != null) {
+                actionAfterDeath(patient, patient.getDeathDate());
```

Comment:
Why not just pass in one parameter?

Reply:
Modified  the function to take in one parameter